src/StreamJsonRpc.Shared/Reflection/MethodNameAttribute.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using Microsoft;
+
+namespace StreamJsonRpc
+{
+    /// <summary>
+    /// Attribute which allows a method to have an alternate name.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = false, Inherited = false)]
```

Comment:
This would make differentiating between base class and derived class super difficult if they have different names for the same method.  Right now I throw an error as soon as I see that a method names to multiple names (MethodNameAttribute values).  If we support multiple = true, the logic to determine whether multiple names for a single method is valid (multiple attributes on the same method in the same class) or invalid (base and derived classes have different values for MethodNameAttribute).  There isn't a real life usage for having to declare multiple names for a method, so I left it as not allowed.